Add warning about derived API keys to docs (#62351)
elastic/elasticsearch: x-pack/docs/en/rest-api/security/create-api-keys.asciidoc

* To use this API, you must have at least the `manage_api_key` cluster privilege.

IMPORTANT: If the credential that is used to authenticate this request is

> an API key, then the derived API key that is created cannot have any privileges.

Reviewer: Does the API return an error if you try to create a derived key with privileges? Maybe it's clearer to say something like this:
```suggestion
an API key, the result is a derived API key that cannot have any privileges. If you specify privileges, the API returns an error.
```

Reviewer 2: I like calling out explicity that an error will be generated if privileges are specified for the derived key. So let's add that sentence. 
But for the sentence precedes it, I prefer Tim's version better because the phrase of "the result is ..." somehow feels like we would internally discard any privileges specified for the derived key and "the result is no privilge", which I personally feels a bit confusing when read together with "If you specify ...".

Reviewer: @ywangd I've altered my suggestion to try to address your comments.

thereby limiting the access scope for API keys.
The structure of role descriptor is the same as the request for create role API.
For more details, see <<security-api-put-role, create or update roles API>>.
+
--
NOTE: Due to the way in which this permission intersection is calculated, it is not
possible to create an API key that is a child of another API key, unless the derived
key is created without any privileges. In this case, you must explicitly specify a
role descriptor with no privileges, and the derived API key can be used for

> authentication only, but will not have access to call {es} APIs.

Reviewer 2: Technically, it is not "only" the authentication, but a few actions that requires "the same user", e.g. it can also call `GET _security/api_key?owner=true`. Without complicating things too much by explaining all the details, maybe we could just drop the `only` bit.

The structure of role descriptor is the same as the request for create role API.
For more details, see <<security-api-put-role, create or update roles API>>.

NOTE: Due to the way in which this permission intersection is calculated, it is not
possible to create an API key that is a child of another API key, unless the derived
key is created without any privileges. In this case, you must explicitly specify a

> role descriptor with no privileges, and the derived API key can be used for

Reviewer: Not mandatory, but I think this could be split into two sentences.
```suggestion
role descriptor with no privileges. The derived API key can be used for
```

> authentication only, but will not have access to call {es} APIs.

Reviewer: I think "but" should be removed here. Should "access" also be changed to "authority"?
```suggestion
authentication only; it will not have authority to call {es} APIs.
```

Reviewer: > Without complicating things too much by explaining all the details, maybe we could just drop the `only` bit.

I've made that change in my suggestion, thanks!
